docs/source/extending/PivotFunctions.rst

```diff
@@ -0,0 +1,270 @@
++-------------------------+-------------------------------+------------+------------+
+| src_func_name           | The name of the function to   | Yes        | -          |
+|                         | wrap                          |            |            |
++-------------------------+-------------------------------+------------+------------+
+| func_new_name           | Rename the function           | No         | -          |
++-------------------------+-------------------------------+------------+------------+
+| input type              | The input type that the       | Yes        | -          |
+|                         | wrapped function expects      |            |            |
+|                         | (DataFrame iterable value)    |            |            |
++-------------------------+-------------------------------+------------+------------+
+| entity_map              | Mapping of entity and         | Yes        | -          |
+|                         | attribute used for function   |            |            |
++-------------------------+-------------------------------+------------+------------+
+| func_df_param_name      | The param name that the       | If DF      | -          |
+|                         | function uses as input param  | input      |            |
+|                         | for DataFrame                 |            |            |
++-------------------------+-------------------------------+------------+------------+
+| func_df_col_param_name  | The param name that function  | If DF      | -          |
+|                         | uses to identify the input    | input      |            |
+|                         | column name                   |            |            |
++-------------------------+-------------------------------+------------+------------+
+| func_out_column_name    | Name of the column in the     | If DF      | -          |
+|                         | output DF to use as a key to  | output     |            |
+|                         | join                          |            |            |
++-------------------------+-------------------------------+------------+------------+
+| func_static_params      | dict of static name/value     | No         | -          |
+|                         | params always sent to the     |            |            |
+|                         | function                      |            |            |
++-------------------------+-------------------------------+------------+------------+
+| func_input_value_arg    | Name of the param that the    | If not     | -          |
+|                         | wrapped function uses for its | DF input   |            |
+|                         | input value                   |            |            |
++-------------------------+-------------------------------+------------+------------+
+| can_iterate             | True if the function supports | No         | Yes        |
```

Comment:
Do pivot functions with `can_iterate=True` behave like this?

```python
# .get_recent_alerts would be the custom pivot function
for recent_alert in IpAddress.get_recent_alerts("157.53.1.1"):
    ...
```

Just trying to understand how they would be used.

Reply:
It really means that the function is not too expensive to run. When you supply an iterable of DF input to pivot functions, if the wrapped (underlying) function only accepts single items, the pivot stuff will call the wrapped function multiple times and concat the results. This is fine for most things but some things like notebooklets are big and expensive and I wanted to avoid iterating over those by accident.